src/ui/src/renderer_widget.cpp

```diff
@@ -82,3 +94,52 @@ QSurfaceFormat surface_format(const QSurfaceFormat::FormatOptions options) {
 	surface_format.setStencilBufferSize(8);
 	return surface_format;
 }
+
+void RendererWidget::mousePressEvent(QMouseEvent* event) { 
+	previous_mouse_pos_ = event->pos();
+	event->accept();
+}
+
+
+void RendererWidget::mouseReleaseEvent(QMouseEvent* event) { 
+	event->accept(); 
+}
+
+void RendererWidget::mouseMoveEvent(QMouseEvent* event) {
+	auto clicked_buttons = event->buttons();
+	// if button still pressed we use it and don't mention other clicked buttons
+	if (current_pressed_button_ & clicked_buttons)
+		clicked_buttons = current_pressed_button_;
+	bool action_requiered = true;
+	const auto q_curr_pos = event->pos();
+	const auto q_delta = q_curr_pos - previous_mouse_pos_;
+	const Vec2D curr_pos(q_curr_pos.x(), q_curr_pos.y());
+	const Vec2D delta(q_delta.x(), q_delta.y());
+	// rotate
+	if ((clicked_buttons & Qt::RightButton) && action_requiered) {
+		current_pressed_button_ = Qt::RightButton;
+		action_requiered = false;
+		viewport_->RotateCamera(curr_pos, delta);
+	}
+	// move
+	if ((clicked_buttons & Qt::MiddleButton) && action_requiered) {
+		current_pressed_button_ = Qt::RightButton;
+		action_requiered = false;
+		viewport_->MoveCamera(curr_pos, delta);
+	}
+	previous_mouse_pos_ = q_curr_pos;
+	event->accept();
+}
+
+void RendererWidget::mouseDoubleClickEvent(QMouseEvent* event) {
+	event->accept();
+}
+
+void RendererWidget::wheelEvent(QWheelEvent* event) { 
+	// TODO play with
+	const float sensivity = 1.0f;
+	const float delta = event->angleDelta().y() * sensivity;
```

Comment:
Narrowing conversion from `int` to `float`. Escape them or make explicit (static_cast them all)